bark/world/evaluation/rss/safety_polygon.hpp

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2020 fortiss GmbH
+//
+// Authors: Julian Bernhard, Klemens Esterle, Patrick Hart and
+// Tobias Kessler
+//
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT>.
+
+#ifndef BARK_WORLD_EVALUATION_SAFETY_POLYGON_HPP_
+#define BARK_WORLD_EVALUATION_SAFETY_POLYGON_HPP_
+
+#include <limits>
+#include <memory>
+#include <string>
+
+#include "bark/world/world.hpp"
+#include "bark/geometry/polygon.hpp"
+#include "bark/geometry/commons.hpp"
+#include "bark/world/world.hpp"
+
+
+namespace bark {
+namespace world {
+namespace evaluation {
+
+using geometry::Polygon;
+using bark::world::AgentId;
+using objects::StateDefinition;
+using bark::geometry::SignedAngleDiff;
+namespace bg = boost::geometry;
+
+// TODO: make serializable
+struct SafetyPolygon {
+  double lat_left_safety_distance{0.};
+  double lat_right_safety_distance{0.};
+  double lon_safety_distance{0.};
+  Polygon polygon;
+  AgentId agent_id{0};
+  double curr_distance{0.};
+  Polygon GetPolygon() const {
+    return polygon;
+  }
+  AgentId GetAgentId() const {
+    return agent_id;
+  }
+};
+
+typedef std::shared_ptr<SafetyPolygon> SafetyPolygonPtr;
+
+inline void ComputeSafetyPolygon(
```

Comment:
what is this function doing? is it transforming a cartesian axis-aligned polygon to the orientation of the vehicle of the agent? can it be implemented using some base functions from bark::polygon (transform or so?)
either way, a doc-string might be beneficial.

Reply:
Okay, will add that. I checked and I did not find any appropriate geometry functions.
boost::geometry::scale_transformer e.g. transforms it in all dimensions.